src/Google.Datastore.V1Beta3/ArrayValuePartial.cs

```diff
@@ -0,0 +1,346 @@
+        /// <summary>
+        /// Converts an array to an <see cref="ArrayValue"/> instance. Each value within <paramref name="values"/> is converted
+        /// to a <see cref="Value"/> using the individual implicit conversion.
+        /// </summary>
+        /// <param name="values">The array to convert.</param>
+        /// <returns>An <see cref="ArrayValue"/> with converted values, or <c>null</c> if <paramref name="values"/> is null.</returns>
+        public static implicit operator ArrayValue(DateTime[] values) => ToArrayValue(values, x => x);
+        /// <summary>
+        /// Converts an array to an <see cref="ArrayValue"/> instance. Each value within <paramref name="values"/> is converted
+        /// to a <see cref="Value"/> using the individual implicit conversion. Any null input values are converted
+        /// to instances of <see cref="Value"/> with a kind of Null.
+        /// </summary>
+        /// <param name="values">The array to convert.</param>
+        /// <returns>An <see cref="ArrayValue"/> with converted values, or <c>null</c> if <paramref name="values"/> is null.</returns>
+        public static implicit operator ArrayValue(DateTime?[] values) => ToArrayValue(values, x => x);
+        /// <summary>
+        /// Converts an array to an <see cref="ArrayValue"/> instance. Each value within <paramref name="values"/> is converted
+        /// to a <see cref="Value"/> using the individual implicit conversion.
+        /// </summary>
+        /// <param name="values">The array to convert.</param>
+        /// <returns>An <see cref="ArrayValue"/> with converted values, or <c>null</c> if <paramref name="values"/> is null.</returns>
+        public static implicit operator ArrayValue(DateTimeOffset[] values) => ToArrayValue(values, x => x);
+        /// <summary>
+        /// Converts an array to an <see cref="ArrayValue"/> instance. Each value within <paramref name="values"/> is converted
+        /// to a <see cref="Value"/> using the individual implicit conversion. Any null input values are converted
+        /// to instances of <see cref="Value"/> with a kind of Null.
+        /// </summary>
+        /// <param name="values">The array to convert.</param>
+        /// <returns>An <see cref="ArrayValue"/> with converted values, or <c>null</c> if <paramref name="values"/> is null.</returns>
+        public static implicit operator ArrayValue(DateTimeOffset?[] values) => ToArrayValue(values, x => x);
+        /// <summary>
+        /// Converts an array to an <see cref="ArrayValue"/> instance. Any null input values are converted
+        /// to instances of <see cref="Value"/> with a kind of Null.
+        /// </summary>
+        /// <param name="values">The array to convert.</param>
+        /// <returns>An <see cref="ArrayValue"/> with converted values, or <c>null</c> if <paramref name="values"/> is null.</returns>
+        public static implicit operator ArrayValue(Value[] values) => ToArrayValue(values, x => x);
```

Comment:
Sorry, didn't realize this before. This one might be a little odd to convert nulls. If you already have an array of Value instances, I'm guessing you just want to copy them directly to the ArrayValue.Values. Although perhaps not. Is it even valid to have nulls in the ArrayValue.Values? If not and it always causes issues, then I guess this is fine.